### PR TITLE
DAOS-14559 ci: Update isort config and add GitHub action for it.

### PR DESCRIPTION
**Pull request by ashleypittman**
### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
@ashleypittman FYI https://github.com/daos-stack/daos/pull/13247

File: .github/workflows/flake.yml [hunk 1]
```diff
@@ -37,3 +42,23 @@ jobs:
           ignore: 'F821,W503,F841'
           max-line-length: '100'
           args: '--filename */SConscript, SConstruct'
+
+  # Run isort on the tree.
+  # This checks .py files only so misses SConstruct and SConscript files are not checked, rather
+  # for these files check them afterwards.  The output-filter will not be installed for this part
+  # so regressions will be detected but not annotated.
+  isort:
+    name: Isort check
```
Comment (daltonbohning) [inline .github/workflows/flake.yml:51]: Does an isort check really belong in `flake.yml`?

File: .github/workflows/flake.yml [hunk 2]
```diff
@@ -37,3 +42,23 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v3
+        with:
+          python-version: 3.11
+      - uses: isort/isort-action@master
```
Comment (daltonbohning) [inline .github/workflows/flake.yml:58]: Is this checked in?

Comment (ashleypittman) [reply .github/workflows/flake.yml:58]: this is a public action, it doesn't need to be checked in.

**Comment by ashleypittman** [comment]
> @ashleypittman FWIW, I have plans to add an ftest-tag lint check at some point. Maybe we're due for a generic "lint" workflow?

What would `ftest-tag` do?

Initially we created lots of GitHub actions jobs and they're all listed at https://github.com/daos-stack/daos/actions even the ones which have only run once so creating lots of jobs with unique names is a bad idea IMO.  It can also take some time to schedule runners so I am thinking it's best if we actually just have fewer actions but each with multiple steps, it would likely schedule faster and therefore run faster although there is a down-side - you can only have 10 annotations per job, and like we see with codespell and the d_log_check if the first one fails then the second doesn't get run, although in that case they're seperate steps but the same job.

**Comment by daltonbohning** [comment]
> > @ashleypittman FWIW, I have plans to add an ftest-tag lint check at some point. Maybe we're due for a generic "lint" workflow?
> 
> What would `ftest-tag` do?
> 
> Initially we created lots of GitHub actions jobs and they're all listed at https://github.com/daos-stack/daos/actions even the ones which have only run once so creating lots of jobs with unique names is a bad idea IMO. It can also take some time to schedule runners so I am thinking it's best if we actually just have fewer actions but each with multiple steps, it would likely schedule faster and therefore run faster although there is a down-side - you can only have 10 annotations per job, and like we see with codespell and the d_log_check if the first one fails then the second doesn't get run, although in that case they're seperate steps but the same job.

It would verify that tests have appropriate tags. e.g.
- each test needs a unique tag
- needs pr, daily, weekly, etc.
- ...

We can discuss this more when I'm ready to add one

File: .github/workflows/flake.yml (hunk 1 above)
Comment (YinglinSun) [reply .github/workflows/flake.yml:51]: maybe renaming this flake workflow to something more generic like `cq.yml`? that way we can include all cq related checks in one file.

**Pull request closed without merging**